app/controllers/backend/fixed_assets_controller.rb

```diff
@@ -200,26 +200,27 @@ def depreciate
 
     protected
 
-    def find_fixed_assets
-      fixed_asset_ids = params[:id].split(',')
-      fixed_assets = fixed_asset_ids.map { |id| FixedAsset.find_by(id: id) }.compact
-      unless fixed_assets.any?
-        notify_error :no_fixed_assets_given
-        redirect_to(params[:redirect] || { action: :index })
-        return nil
+      def find_fixed_assets
+        fixed_asset_ids = params[:id].split(',')
+        fixed_assets = fixed_asset_ids.map { |id| FixedAsset.find_by(id: id) }.compact
+        unless fixed_assets.any?
+          notify_error :no_fixed_assets_given
+          redirect_to(params[:redirect] || { action: :index })
+          return nil
+        end
+        fixed_assets
       end
-      fixed_assets
-    end
 
 
-    def parameters_with_processed_percentage
-      parameters = permitted_params.dup
-      parameters['depreciation_percentage'] = case parameters['depreciation_method']
-        when 'linear' then parameters.delete('linear_depreciation_percentage')
-        when 'regressive' then parameters.delete('regressive_depreciation_percentage')
-        else ''
+      def parameters_with_processed_percentage
+        parameters = permitted_params.to_h
+        method = parameters.fetch('depreciation_method', nil)
+        if method
+          percentage_key = "#{method}_depreciation_percentage"
+          depreciation_percentage = parameters.fetch(percentage_key, '')
+          parameters['depreciation_percentage'] = depreciation_percentage if depreciation_percentage.to_s != ''
```

Comment:
`depreciation_percentage.present?` or `unless ... .blank?` would be more concise here